src/Stack2nix.hs

```diff
@@ -234,11 +237,32 @@ toNix Args{..} remoteUri baseDir StackConfig{..} =
             Success expr ->
               case takeFileName fname of
                 "hspec.nix" ->
-                  writeFile fname $ show $ prettyNix $ (addParam "stringbuilder" . stripNonEssentialDeps) expr
-                _ ->
-                  writeFile fname $ show $ prettyNix $ stripNonEssentialDeps expr
+                  writeFile fname $ show $ prettyNix $ (addParam "stringbuilder" . stripNonEssentialDeps False) expr
+                name ->
+                  case argTestPkg of
+                    Just pkg -> do
+                      let shouldTest = name == pkg <.> "nix"
```

Comment:
If I understand correctly we could use `packages` function here and compare it to `name` to support just `--test` that would enable tests for all pacakges in `packages` directive (minus those filtered if `extra-dep: true`).